### ISSUE TITLE
Cannot use with IEnumVARIANT interface and _NewEnum property

### ISSUE THREAD

**Issue by dtylman** [go-ole/go-ole#99]
Trying to implement WMI query and enumerate all objects returned.

The following code crashes.

```go

func main(){
	defer func(){
		r:=recover()
		if r!=nil{
			fmt.Printf("%v",r)
		}
	}()
	mutex:=sync.Mutex{}
	mutex.Lock()
	defer mutex.Unlock()
	runtime.LockOSThread()
	defer runtime.UnlockOSThread()

	err := ole.CoInitializeEx(0, ole.COINIT_APARTMENTTHREADED)
	if err != nil {
		code := err.(*ole.OleError).Code()
		fmt.Printf("CoInitializeEx returned with %v", err)
		if code == 0x00000001 { // The COM library is already initialized on this thread.

		} else {
			panic(err)
		}
	}else {
		defer ole.CoUninitialize()
	}

	comserver, err := oleutil.CreateObject("WbemScripting.SWbemLocator")
	if err != nil {
		fmt.Printf("CreateObject WbemScripting.SWbemLocator returned with %v", err)
		panic(err)
	}
	defer comserver.Release()
	dispatch,err:= comserver.QueryInterface(ole.IID_IDispatch)
	if err != nil {
		fmt.Printf("context.iunknown.QueryInterface returned with %v", err)
		panic(err)
	}
	defer dispatch.Release()
	wbemServices := oleutil.MustCallMethod(dispatch, "ConnectServer")
	defer wbemServices.Clear()
	wbemServices_dispatch:= wbemServices.ToIDispatch()
	defer wbemServices_dispatch.Release()

	objectset:= oleutil.MustCallMethod(wbemServices_dispatch,"ExecQuery","SELECT * FROM WIN32_Process")
	objectset_dispatch:=objectset.ToIDispatch()
	defer objectset.Clear()
	variant:=oleutil.MustGetProperty(objectset_dispatch,"_NewEnum")
	defer variant.Clear()
	object2:=variant.ToIUnknown()
	defer object2.Release()	
	
	//Not sure what to do here:
	enum:=ole.IEnumVARIANT{*object2}
	defer enum.Release()

	a,l,err:=enum.Next(1)
	if err!=nil{
		panic(err)
	}
	fmt.Println(&a)	
	fmt.Println(l)
}
```

**Comment by jacobsantos** [comment]
```go
dispatch,err:= comserver.QueryInterface(ole.IID_IDispatch)
```

`ole.IID_IDispatch` has to be the IID or CLSID of the object you are referencing for this to work.

**Comment by jacobsantos** [comment]
https://msdn.microsoft.com/en-us/library/aa393719(v=vs.85).aspx

I am attempting to get the value of the GUID, but I might just need to look at the registry.

**Comment by dtylman** [comment]
I think its `var IID_IEnumVariant = &ole.GUID{0x00020404,0x0000,0x0000,[8]byte{0xC0,0x00,0x00,0x00,0x00,0x00,0x00,0x46}}`

**Comment by jacobsantos** [comment]
See #101

**Comment by jacobsantos** [comment]
Yeah, so the test case should provide a bit more information. Type `go test` in go-ole checkout and see what the error is. If you can paste that error, then we can continue.

```
        idispatch_windows_test.go:32: Class not registered
Exception 0xc0000005 0x8 0x100009 0x100009 PC=0x100009

syscall.Syscall(0x7fffb3701810, 0x1, 0xc0820072a0, 0x0, 0x0, 0xc08201eb30, 0x43362d, 0xc08201600c)
       c:/go/src/runtime/syscall_windows.go:128 +0x5c 
                syscall.(*Proc).Call(0xc082006900, 0xc08202fab8, 0x1, 0x1, 0x0, 0x6a1be8, 0x0, 0x0)
       c:/go/src/syscall/dll_windows.go:130 +0x18e 
                _/golang/go-ole/go-ole-fork.VariantClear(0xc0820072a0, 0x0, 0x0)
       c:/golang/go-ole/go-ole-fork/com.go:223 +0x8c
                _/golang/go-ole/go-ole-fork.(*VARIANT).Clear(0xc0820072a0, 0x0, 0x0)
       c:/golang/go-ole/go-ole-fork/variant.go:45 +0x34
                reflect.Value.UnsafeAddr(0x5a36e0, 0x2d9480, 0x16, 0x2d9480)
       c:/go/src/reflect/value.go:1723 +0xd4
                _/golang/go-ole/go-ole-fork.reflectQueryInterface(0x5a36e0, 0x2d9480, 0x7fffa34bc7c0, 0xc08202fd78, 0x53bae0, 0xc082002e30, 0x0, 0x0)
       c:/golang/go-ole/go-ole-fork/iunknown_windows.go:15 +0x8c
                _/golang/go-ole/go-ole-fork.(*IUnknown).PutQueryInterface(0x2d9480, 0xc08202fd78, 0x538660, 0xc082028078, 0x0, 0x0)
       c:/golang/go-ole/go-ole-fork/iunknown.go:26 +0xb9
                _/golang/go-ole/go-ole-fork.(*IUnknown).IEnumVARIANT(0x2d9480, 0xc08202fd78, 0x0, 0x0, 0x0)
       c:/golang/go-ole/go-ole-fork/iunknown.go:35 +0x82
                _/golang/go-ole/go-ole-fork.TestIEnumVariant_wmi(0xc08200c870)
       c:/golang/go-ole/go-ole-fork/ienumvariant_test.go:90 +0x10fb
                testing.tRunner(0xc08200c870, 0x69f398)
       c:/go/src/testing/testing.go:456 +0x9f
created by testing.RunTests
       c:/go/src/testing/testing.go:561 +0x874
```

Is the error I received. I suppose it is possible that a class ID is not registered with WMI. I need to check it more.

**Comment by jacobsantos** [comment]
Appears to be fixed, per comment in #101
